app/controllers/user/dashboard_controller.rb

```diff
@@ -0,0 +1,8 @@
+class User::DashboardController < ApplicationController
+  before_action :require_current_user
+  def show
+    @npcs = UserNpcModel.where(user_id: current_user.id).map do |user_npc_model|
+      NpcModel.where(id: user_npc_model.npc_model_id)
```

Comment:
I'm only generating the NPCs here and not their weapons and everything because the Dashboard show page is only currently using their name.